mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -185,6 +186,56 @@ void AbstractControllerExecution::setVelocityCmd(const geometry_msgs::TwistStamp
   // TODO so there should be no loss of information in the feedback stream
 }
 
+bool AbstractControllerExecution::checkCmdVelIgnored(const geometry_msgs::Twist& cmd_vel)
+{
+  // check if the velocity ignored check is enabled or not
+  if (cmd_vel_ignored_tolerance_ <= 0.0)
+  { 
+    return false;
+  }
+
+  const bool robot_stopped = robot_info_.isRobotStopped(1e-3, 1e-3);
+
+  // compute linear and angular velocity magnitude
+  const double cmd_linear = std::hypot(cmd_vel.linear.x, cmd_vel.linear.y);
+  const double cmd_angular = std::abs(cmd_vel.angular.z);
+
+  const bool cmd_is_zero = cmd_linear < 1e-3 && cmd_angular < 1e-3;
+
+  if (!robot_stopped || cmd_is_zero)
+  {
+    // velocity is not being ignored
+    if (!first_ignored_time_.is_zero())
+    {
+      first_ignored_time_ = ros::Time();
+    }
```

Comment:
```suggestion
    first_ignored_time_ = ros::Time();
```
F: Remove the if statement since it's not doing anything -- we always enter it.

Reply:
It is not always the situation that we will enter this if statement. When cmd_vel is not being ignored last time and also this time, first_ignored_time_ is already set to zero, so it would not enter this if statement.

Reply:
Yes, you're right. I didn't notice you were calling `ros::Time()` instead of `ros::Time::now()`.

However, I would still keep my suggestion here, for a different reason now.
There is no need to check if the time is zero -- you can just always reset it.
That makes the code easier to read.